kakarot_scripts/utils/kakarot.py

```diff
@@ -55,10 +57,74 @@
 from tests.utils.constants import TRANSACTION_GAS_LIMIT
 from tests.utils.helpers import pack_calldata, rlp_encode_signed_data
 
-logging.basicConfig()
+logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+_nonces = {}
+
+
+async def get_nonce(account):
+    global _nonces
+    if account.address not in _nonces:
+        if WEB3.is_connected():
+            _nonces[account.address] = WEB3.eth.get_transaction_count(
+                account.signer.public_key.to_checksum_address()
+            )
+        else:
+            _nonces[account.address] = (
+                await (
+                    _get_starknet_contract("account_contract", address=account.address)
+                    .functions["get_nonce"]
+                    .call(block_number="pending")
+                )
+            ).nonce
+
+    if WEB3.is_connected():
+        network_nonce = WEB3.eth.get_transaction_count(
+            account.signer.public_key.to_checksum_address()
+        )
+    else:
+        network_nonce = (
+            await (
+                _get_starknet_contract("account_contract", address=account.address)
+                .functions["get_nonce"]
+                .call(block_number="pending")
+            )
+        ).nonce
+
+    retries = 10
+    while network_nonce != _nonces[account.address] and retries > 0:
+        logger.info(
+            f"⏳ Waiting for network nonce {network_nonce} to be {_nonces[account.address]}"
```

Comment:
c'est pour solve quoi exactement ça ? attendre que les tx soient "minées"?

Reply:
because the state got from the RPC node might not be up to date with the latest tx, I was facing a lot of nonce mismatch so I decided to take this conservative approach here: waiting for retrieved nonce to be the one expected, or wait for 1s